### ISSUE TITLE
chemical search (5 pts)

### ISSUE THREAD

**Issue by kdionisio** [HumanExposure/factotum#274]
Dependent on #272 and #285 

As an EPA colleague
I want to see a summary of the data documents associated with a specific chemical.
So that I know how much information Factotum might be able to provide me on a chemical of interest.

From the Dashboard page, user should have the ability to look-up data on a chemical (by chemical name, CAS, or DTXSID). Query should search both the dsstoxsubstance data model, and the extractedchemical data model.

Query should return the number of data documents associated with that chemical, split by number of data documents associated with that chemical through the dsstoxsubstance data model, and the number of data documents associated with that chemical through the extractedchemical data model. For the latter, any data document which is included as being linked through the dsstoxsubstance data model should NOT be included in the count for documents linked through the extractedchemical data model (i.e., a single data document should only be counted once).

When the numbers are returned, they can be labeled on the interface as:
- "Number of documents with substance registered in DSSTox" (this would be the number of documents associated with chemicals that have a DTXRID present in the dsstoxsubstance table)
- "Number of documents potentially associated with chemical substance" (this would be the number of documents associated with chemicals through the extractedchemical data model)

Definition of Done:
Chemical search available on Dashboard which returns # data documents by dsstoxsubstance and # data documents by extractedchemical, where documents are not counted twice

**Comment by kdionisio** [comment]
Note: @mwfrost will update/suggest updates, ticket to include only creation of query and NOT interface functionality

**Comment by mwfrost** [comment]
Proposed rewrite of ticket:

# Add dsstoxsubstance and extractedchemical objects to search index 

Dependent on #272 

As a developer
I want to be able to add a search box for finding data documents associated with a specific chemical to the dashboard page.

The search engine should return data on a chemical (by chemical name, CAS, or DTXSID) based on a query string. Query should search both the dsstoxsubstance data model and the extractedchemical data model.

Query should return the number of data documents associated with that chemical, split by number of data documents associated with that chemical through the dsstoxsubstance data model, and the number of data documents associated with that chemical through the extractedchemical data model. For the latter, any data document which is included as being linked through the dsstoxsubstance data model should NOT be included in the count for documents linked through the extractedchemical data model (i.e., a single data document should only be counted once).

The summary numbers returned will eventually be labeled on the interface as:
- "Number of documents with substance registered in DSSTox" (this would be the number of documents associated with chemicals that have a DTXRID present in the dsstoxsubstance table)
- "Number of documents potentially associated with chemical substance" (this would be the number of documents associated with chemicals through the extractedchemical data model)

Definition of Done:
A view which returns, based on a chemical name query string, # data documents by dsstoxsubstance and # data documents by extractedchemical, where documents are not counted twice

Test:
A new view returns both formatted lists of data documents in response to a search term and summary counts of the documents

**Comment by kdionisio** [comment]
First - approve of proposed ticket rewrite by @mwfrost .
Yes, @Daveahern we are only focusing on back-end piece for now. I will write another ticket for the front end piece of this.

**Comment by Daveahern** [comment]
@kdionisio how smart do you want the chemical search to be?  In our sample data we have some extracted chemical records with raw chem names of "ethylparaben", "ethylparaben_P" and "ethylparaben_PP".  The standardized DSSToxSubstance record has a true_chemname of "ethylparaben"  for all three records. Elastic search is smart enough that if you search for "ethylparaben_P" it will return to you all three of these records, even though that value only occurs in one of them (in the raw chem name).  So using elastic search, we have this sort of "fuzzy" matching. The chem search will then return to you a list saying we have three data documents with matching DSSTox records in them, even though the search string entered was misspelled. 

Additionally, if you enter a search string of "dibutyl ethylparaben" it will return to you a count of the documents that contain one or the other or both of those chemical names.

Is this how smart you want the search, or do you want something that's more exacting?
